src/Core/Commander/Providers/Fetcher.js

```diff
@@ -0,0 +1,41 @@
+import { TextureLoader } from 'three';
+
+let textureLoader = new TextureLoader();
+
+function checkResponse(response) {
+    if (response.status < 200 || response.status >= 300) {
+        var error = new Error(`Error loading ${response.url}: status ${response.status}`);
+        error.status = response.status;
+        throw error;
+    }
+}
+
+let Fetcher = {
+
+    json: function(url) {
+        return fetch(url).then(response => {
+            checkResponse(response);
+            return response.json();
+        });
+    },
+
+    xml: function(url) {
+        return fetch(url).then(response => {
+            checkResponse(response);
+            return new window.DOMParser().parseFromString(response.text(), 'text/xml');
+        });
+    },
+
+    texture: function(url) {
+        let res, rej;
+        let promise = new Promise(function(resolve, reject) {
+            res = resolve;
+            rej = reject;
+        });
+
+        let texture = textureLoader.load(url, res, () => {}, rej);
+        return {texture, promise};
+    }
+}
+
+export default Fetcher;
```

Comment:
Instead of `export default Fetcher` how about exporting each function instead?

```js
export function json(url) {
   …
}
export function xml(url) {
   …
}
export function texture(url) {
   …
}
```